pkg/operator/options/options.go

```diff
@@ -74,6 +74,10 @@ type Options struct {
 	SubnetID string // => VnetSubnetID to use (for nodes in Azure CNI Overlay and Azure CNI + pod subnet; for for nodes and pods in Azure CNI), unless overridden via AKSNodeClass
 	setFlags map[string]bool
 
+	ManagedKarpenter bool // => ManagedKarpenter is true if Karpenter is managed by AKS, false if it is a self-hosted karpenter installation
```

Comment:
remind me: is there any way to prevent self-hosted users from setting this?

Reply:
no, presumably they good set it if they wanted to, but after we put Agentbaker bootstrapping behind this flag bootstrapping won't work for them. 

 I guess we could also determine if we are using managed based on the kubeclients or build path(main_ccp.go vs main.go) but i thought we wanted to remove those and consolidate them eventually.

Reply:
Maybe this should be an enum?

Mode:
- Managed
- SelfHosted

With default SelfHosted?

@matthchr, feel like I remeber best practice of using enums instead of bools?